### DISCUSSION TITLE
import data

### DISCUSSION THREAD

**Discussion by FlorianBeckOle** [nextpyp/.github#6] [Q&A] [answered]
I can only browse inside the users Directory for data import.
e.g /data/projectsNext/users/user1/

If I try to go one level up I get 
"folder not allowed "message.

For space and security reasons It would be good,
if the user could also import data from a different 
directory(s).

Is that possible ? What do I need to configure.

thanks

Florian

**Comment by abartesaghi** [comment, marked as answer]
Florian, since nextPYP runs inside of a Singularity/Apptainer container, by default, no files from outside of the container will be visible once inside the application. To make files visible, you need to bind the directories containing those files into the container using the `binds` parameter in the `[pyp]` section of the `config.toml` configuration file. This is explained in Step 5 of the [installation](https://nextpyp.app/files/pyp/latest/docs/install/install-web.html#step-5-recommended-configure-access-to-system-resources). 

Depending on what you want to make visible, it looks like you will need to add something like the following to your configuration file:
```
[pyp]
binds = [ '/data' ]
```
For the change to take effect, you will need to restart nextPYP.

**Reply by FlorianBeckOle** [reply]
Thanks that works.


best
Florian

**Reply by kexinliu0001** [reply]
hi,

I binded the directories, and tried to restart nextpyp.
but the start process is failed.

how can I overcome it?
![Uploading WechatIMG127.jpg…]()
![Uploading WechatIMG128.jpg…]()

**Comment by yuhx47** [comment]
Hi,

I always encountered the internal server error on the top right corner when I import raw tutorial data, 
And the job stuck there. 
how can I overcome the error?

<img width="325" alt="image" src="https://github.com/nextpyp/.github/assets/37691985/a0092b94-abac-4ca9-a75b-165da5885e30">

Thanks, 
Huaxin

**Reply by cuchaz** [reply]
The "Internal server error" message in the browser means there's another more detailed error message in the server `micromon` log. The log should be in your `local/logs` folder. [See the installation docs for more info about the log files.](https://nextpyp.app/files/pyp/latest/docs/install/install-web.html#log-files) There will probably be more information about the error in that log. If you want to share it here, we can try to figure out what went wrong.

**Comment by yuhx47** [comment]
[micromon.txt](https://github.com/nextpyp/.github/files/13273254/micromon.txt)
Please see the attached file. 

Thanks

**Reply by cuchaz** [reply]
The error log says the website can't connect to your SLURM login node using the `slurm.host` config value in your `config.toml` file. Generally, that value should be the hostname of your SLURM login node. If you don't have a SLURM cluster handy, just delete the whole `[slurm]` section from your config file to turn on standalone mode.

There are some configuration options the installer can't guess correctly, so you'll have to edit the `config.toml` file to put in the correct values for your environment. [See this part of the installation instructions for more details.](https://nextpyp.app/files/pyp/latest/docs/install/install-web.html#step-4-review-configuration) If the installation instructions don't help you get this working, let us know if you have any more questions.

**Reply by yuhx47** [reply]
Once I deleted the whole SLURM section, I got the eOUTPUT when I restarted using 

sudo -u yu PYP_CONFIG=/home/yu/Programs/nextPYP/config.toml nextpyp-start

Output:

Using configuration at /home/yu/Programs/nextPYP/config.toml
Reading config.toml using CLI tool ...
Exception in thread "main" edu.duke.bartesaghi.micromon.TomlParseException: missing field "user"
	at edu.duke.bartesaghi.micromon.IoKt.getStringOrThrow(io.kt:497)
	at edu.duke.bartesaghi.micromon.IoKt.getStringOrThrow$default(io.kt:496)
	at edu.duke.bartesaghi.micromon.Config.<init>(Config.kt:186)
	at edu.duke.bartesaghi.micromon.Config$Companion.fromCanon(Config.kt:31)

**Reply by yuhx47** [reply]
Now I successfully started it when I added the following to config.toml, 

[standalone]
availableCpus = 4

but this time I was told I am not an admin user after starting the web interface.

**Reply by cuchaz** [reply]
When deleting the `slurm` section, make sure you delete the `[slurm]` text from the config file as well, or the website will still try to look for the SLURM configuration.

Once your instance is started, you can setup your admin user by following the [first-time setup instructions](https://nextpyp.app/files/pyp/latest/docs/install/install-web.html#step-8-first-time-setup).

**Reply by yuhx47** [reply]
I met a situation as following screenshot, 
actually, I re-installed the program this time, I set up the admin and users successfully before, 
could you give me some advice to find the admin information of the nextPYP settings?

<img width="1379" alt="image" src="https://github.com/nextpyp/.github/assets/37691985/8e7fdfe8-fdfd-449d-9729-d27406eb0d92">

**Reply by cuchaz** [reply]
It looks like you've already done the first time setup? Try logging in as the user account you created during the setup. There's a login link in the upper right corner of your screenshot. That login account should be an administrator.

If you're not able to log in, then try deleting the database (delete the entire `local/db` folder), restarting the server, and going through the first-time setup again.

**Reply by yuhx47** [reply]
Thanks, 

<img width="689" alt="image" src="https://github.com/nextpyp/.github/assets/37691985/0860e924-94b7-49ef-9e21-4e74382c2e02">

It works, I started and logged in,
but as the screenshot shows, the import never finished, the gear keeps running,
and by the way, where is the output written? It seems I did not set the work directory to write files.

**Reply by cuchaz** [reply]
If your job gets stuck, try canceling it (red button on the right) and running it again. If the job still finishes, but fails somehow, you can try to read logs in the browser first. That blue question mark button on the right will change into a file-looking icon and clicking that button will show logs for that job.

If the job never finishes, try reading [the server logs](https://nextpyp.app/files/pyp/latest/docs/install/install-web.html#log-files), particularly the `micromon` log and the `hostprocessor` log.

And definitely make sure your `local` and `shared` folders are writable by the user running the nextpyp process.
